snippets/product-variant-options.liquid

```diff
@@ -0,0 +1,81 @@
+    - variants_option3_arr: {Object} product variants option3 array map.
+
+
+    Usage:
+    {% render 'product-variant-options',
+      product: product,
+      option: option,
+      block: block,
+      variants_available_arr: variants_available_arr,
+      variants_option1_arr: variants_option1_arr,
+      variants_option2_arr: variants_option2_arr,
+      variants_option3_arr: variants_option3_arr
+    %}
+{% endcomment %}
+{%- liquid
+
+  assign product_form_id = 'product-form-' | append: section.id
+
+  assign button_options_html = ''
+  assign dropdown_options_html = ''
+  
+-%}
+
+{%- for value in option.values -%}
+  {%- liquid
+    assign option_disabled = true
+    for option1_name in variants_option1_arr
+      case option.position
+        when 1
+          if variants_option1_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+            assign option_disabled = false
+          endif
+        when 2
+          if option1_name == product.selected_or_first_available_variant.option1 and variants_option2_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+            assign option_disabled = false
+          endif
+        when 3
+          if option1_name == product.selected_or_first_available_variant.option1 and variants_option2_arr[forloop.index0] == product.selected_or_first_available_variant.option2 and variants_option3_arr[forloop.index0] == value and variants_available_arr[forloop.index0]
+            assign option_disabled = false
+          endif
+      endcase
+    endfor
+  -%}
+
+  {%- capture button_options_html -%}
+    {{ button_options_html }}
+    <input type="radio" id="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}"
+          name="{{ option.name }}"
+          value="{{ value | escape }}"
+          form="{{ product_form_id }}"
+          {% if option.selected_value == value %}checked{% endif %}
+          {% if option_disabled %}class="disabled" aria-label="Variant {{ 'products.product.sold_out' | t }} or {{ 'products.product.unavailable' | t }}"{% endif %}
+    >
+    <label for="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}">
+      {{ value }}
+    </label>
+  {%- endcapture -%}
+
+  {%- capture dropdown_options_html -%}
+    {{ dropdown_options_html }}
+    <option value="{{ value | escape }}"{% if option.selected_value == value %} selected="selected"{% endif %}{% if option_disabled and 'disabled' == block.settings.select_soldout_type %} disabled aria-label="Variant {{ 'products.product.sold_out' | t }} or {{ 'products.product.unavailable' | t }}{% endif %}>
+      {{ value }}{% if option_disabled and 'append' == block.settings.select_soldout_type %} [{{ 'products.product.unavailable' | t }}]{% endif %}
+    </option>
+  {%- endcapture -%}
+{%- endfor -%}
+
+{%- if block.settings.picker_type == 'button' -%}
+  {{ button_options_html }}
+{%- elsif block.settings.picker_type == 'dropdown' -%}
+  {{ dropdown_options_html }}
+{%- endif -%}
```

Comment:
I think it would be better to have the `{%- for -%}` output the HTML directly instead of keeping it on a variable that will only be used once. Something like this.

For simplicity and general readability, we usually try to use less variables in Liquid code. If you look at other examples in Dawn, we try to restrict the use of variables to places where it's actually needed for reusability purposes - and even then, only if it's a big block of code that would cause readability issues if repeated too many times.

Thoughts, @ludoboludo?

```suggestion
  {%- if block.settings.picker_type == 'button' -%}
    <input type="radio" id="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}"
          name="{{ option.name }}"
          value="{{ value | escape }}"
          form="{{ product_form_id }}"
          {% if option.selected_value == value %}checked{% endif %}
          {% if option_disabled %}class="disabled" aria-label="Variant {{ 'products.product.sold_out' | t }} or {{ 'products.product.unavailable' | t }}"{% endif %}
    >
    <label for="{{ section.id }}-{{ option.position }}-{{ forloop.index0 }}">
      {{ value }}
    </label>
  {%- elsif block.settings.picker_type == 'dropdown' -%}
    <option value="{{ value | escape }}"{% if option.selected_value == value %} selected="selected"{% endif %}{% if option_disabled and block.settings.select_soldout_type == 'disabled' %} disabled aria-label="Variant {{ 'products.product.sold_out' | t }} or {{ 'products.product.unavailable' | t }}{% endif %}>
      {{ value }}{% if option_disabled and block.settings.select_soldout_type == 'append' %} [{{ 'products.product.unavailable' | t }}]"{% endif %}
    </option>
  {%- endif -%}
{%- endfor -%}
```

Reply:
I like that 👍 Easier to understand I agree.